app/src/main/java/ch/epfl/polybazaar/Utilities.java

```diff
@@ -20,6 +25,10 @@
 
 public abstract class Utilities {
 
+    public static <S, R> Task<R> taskMap(Task<S> t, Callable<R> transform) {
+        return t.onSuccessTask((s) -> Tasks.call(transform));
+    }
+
     public static Map<String, Object> getMap(Object o) {
         Map<String, Object> result = new HashMap<String, Object>();
         if (o instanceof Listing) {
```

Comment:
Couldn't this method be refactored as we have already experienced that it is non-intuitive to come and change the Utilities file when adding a new field to a model

Reply:
Yes, this will be deleted in the next PR